src/main/java/io/confluent/connect/jdbc/source/TableMonitorThread.java

```diff
@@ -105,6 +124,9 @@ public void run() {
     if (tables == null) {
       throw new ConnectException("Tables could not be updated quickly enough.");
     }
+    if (!duplicates.isEmpty()) {
+      throw new ConnectException(configErrorMsg + duplicates.values());
```

Comment:
Can we not just compute `configErrorMsg` inside this block? Why do we want/need to precompute it in the constructor? We still need to hold onto the `duplicates` in a field, but this error would only be thrown once by the `run` method. I'd suggest moving that code into this block to eliminate the `configErrorMsg` field and instead use a local variable.